lib/sensors/temperature.cpp

```diff
@@ -0,0 +1,41 @@
+#include "temperature.hpp"
+
+namespace hyped::sensors {
+
+Temperature::Temperature(hyped::io::I2c &i2c, hyped::core::ILogger &log) : log_(log), i2c_(i2c)
+{
+}
+
+std::optional<std::uint16_t> Temperature::read()
+{
+  if (i2c_.readByte(TEMPERATURE_1, STATUS) == 0) {
+    const std::optional<std::uint8_t> Tempurature_High = i2c_.readByte(kMUX, TEMP_H);
+    const std::optional<std::uint8_t> Tempurature_Low  = i2c_.readByte(kMUX, TEMP_L);
+    // getting the high and low values from the register
+
+    if (Tempurature_High = !std::nullopt, Tempurature_Low = !std::nullopt) {
+      std::optional<std::uint16_t> Tempurature = (Tempurature_High << 8) | Tempurature_Low;
+      Tempurature                              = Tempurature * 0.01;
+
+      return Tempurature;
+    } else {
+      log_.log(hyped::core::LogLevel::kFatal, "Temperature read not implemented");
+    }
+    // Reading from the Status register to see if the tempurature is ready to be read
+    // Need to have it do the correct effect and output a log of what occurred
+  }
+  // log_.log(hyped::core::LogLevel::kFatal, "Temperature read not implemented");
+}
+
+bool Temperature::configure()
+{
+  hyped::core::Result::Check = i2c_.writeByte(kMUX, CTRL, 1);
```

Comment:
hyped::core::Result is a type 

you want `core::Result check = ....` 

check isn't a good var name either so make it more descriptive